packages/@aws-cdk/cx-api/lib/assembly/cloud-assembly-api.ts

```diff
@@ -0,0 +1,56 @@
+import { AppRuntime, MetadataEntry, MissingContext } from '../cxapi';
+import { AssetMetadataEntry } from '../metadata/assets';
+
+export interface ICloudAssembly {
+  readonly directory: string;
+  readonly version: string;
+  readonly missing?: { [key: string]: MissingContext };
+  readonly runtime: AppRuntime;
+  readonly artifacts: ICloudArtifact[];
+  readonly stacks: ICloudFormationStackArtifact[];
+}
+
+export interface ICloudArtifact {
+  readonly assembly?: ICloudAssembly;
+  readonly id: string;
+  readonly environment: Environment;
+  readonly metadata: { [path: string]: MetadataEntry[] };
+  readonly missing: { [key: string]: MissingContext };
+  readonly autoDeploy: boolean;
+  readonly depends: ICloudArtifact[];
+  readonly messages: SynthesisMessage[];
+}
+
+export interface SynthesisMessage {
+  readonly level: SynthesisMessageLevel;
+  readonly id: string;
+  readonly entry: MetadataEntry;
+}
+
+export enum SynthesisMessageLevel {
+  INFO = 'info',
+  WARNING = 'warning',
+  ERROR = 'error'
+}
+
+export interface ICloudFormationStackArtifact extends ICloudArtifact {
+  name: string;
```

Comment:
Mutable? Is this for stack renames? I would rather do without stack renames altogether, really.

Reply:
Let's consider this in a subsequent PR: https://github.com/awslabs/aws-cdk/issues/2670